pandas/core/arrays/integer.py

```diff
@@ -409,8 +409,7 @@ def astype(self, dtype, copy=True):
 
         # if we are astyping to an existing IntegerDtype we can fastpath
         if isinstance(dtype, _IntegerDtype):
-            result = self._data.astype(dtype.numpy_dtype,
-                                       casting='same_kind', copy=False)
```

Comment:
@jreback do you remember what the reason was to explicitly put `casting='same_kind'` ?

Reply:
iirc was to make sure we were
only coercing actual ints (and would raise on floats)

Reply:
But since we have the `if isinstance(dtype, _IntegerDtype):`, we are sure about that? (so it can never be a float dtype)

Reply:
it might have been an artificat.